### ISSUE TITLE
Define `onyo_set()`/`onyo set` behavior for empty keys/values

### ISSUE THREAD

**Issue by TobiasKadelka** [psyinfra/onyo#530]
There are three szenarios in which onyo set can be called with missing keys/values, and they do not behave consistent:

1. both missing, just an `-k =` or `-k ""=` or `-k =""` is passed to satisfy the -k requirement.
```
❱ onyo set -k =
--- /Users/tkadelka/onyo/a_a_a.fauxKJG3b9
+++ /Users/tkadelka/onyo/a_a_a.fauxKJG3b9
@@ -3,3 +3,4 @@
 model: a
 serial: fauxKJG3b9
 key: ''
+'': ''
```

2. no key, but a value is specified:
```
❱ onyo set -k ""=value
zsh: value not found
```

3. just a key, no value, is specified:
```
❱ onyo set -k field=""
--- /Users/tkadelka/onyo/a_a_a.fauxKJG3b9
+++ /Users/tkadelka/onyo/a_a_a.fauxKJG3b9
@@ -3,3 +3,4 @@
 model: a
 serial: fauxKJG3b9
 key: ''
+field: ''
```

I think the szenarios 1.+2. should be just not allowed because they miss a key. I do not see any practical value not having a key.
The third szenario is the only one that might have a reasonable use-case, though even there I think we should discuss if this is wanted behavior.

**Comment by TobiasKadelka** [comment]
The same is also true for `onyo unset --keys ""`, which currently also works.
Of course, as long as `onyo set --keys ""=""` works, there is a need to have an unset-command, too.
But if we decide to define this as illegal for the inventory system, we should add checks for set/unset together.

**Comment by aqw** [comment]
Thank you for writing this up. I was ambivalent about whether we should follow the YAML spec and allow empty keynames, or if we should disallow it for the sake of sanity, typos, etc. The examples you provided made this an easier decision.

Scenario `#2` is `zsh` throwing an error, and there's nothing we can do about that; it's erroring before Onyo even has a chance to see it.

I agree that situations `#1` and `#2` should both behave the same way (i.e. either work or error). And since `zsh` is forcing our hand here with `#2` (which IMP would be the dominant method for declaring this at the CLI), that means we should disallow it also for `#1`.

`#3` we should allow. There's a legitimate use-case for having a key, and wanting the value to be empty.

In short:
- empty `key` is not allowed
- empty `value` is allowed

**Comment by aqw** [comment]
Noting here the conversation we had earlier:

Example 2 is throwing an error because there's a space before the unquoted `=`. I would expect the following to both work:
```
❱ onyo set -k '=value'
```
```
❱ onyo set -k '""=value'
```

In any case, though, I have come down on the side of not allowing empty key names. I find it confusing, and cannot imagine any (sane) use-case.

**Comment by bpoldrack** [comment]
I agree on the conclusion

> In any case, though, I have come down on the side of not allowing empty key names. I find it confusing, and cannot imagine any (sane) use-case.

However, I do want to point out for the record that the way of showing the claimed behavior in OP isn't correct b/c of incorrect quoting. All we see there is the shell's struggling with interpreting the `=` correctly. However, the shell should not interprete the `=` at all. The entire `key=value` string (in all variations) is in fact supposed to be the value of the `-k` option and onyo/argparse needs to get that string as a whole with no shell interference whatsoever. Hence, any error coming from zsh indicates wrong quoting and nothing about onyo.

```
❱ onyo set -k "=empty" -p t_m_mo.fauxfX7ezH                                                                                                                                     
The following assets will be changed:
--- /tmp/test/t_m_mo.fauxfX7ezH
+++ /tmp/test/t_m_mo.fauxfX7ezH
@@ -3,3 +3,4 @@
 make: m
 model: mo
 serial: fauxfX7ezH
+'': empty
Update assets? (y/n) y
```
